src/custom/state/orders/updater.ts

```diff
@@ -128,6 +133,73 @@ const constructGetLogsRetry = (provider: Web3Provider) => {
   return getLogsRetry
 }
 
+export function EventUpdaterApiOnly(): null {
+  const { chainId, library } = useActiveWeb3React()
+
+  const pending = usePendingOrders({ chainId })
+  const fulfillOrdersBatch = useFulfillOrdersBatch()
+
+  useEffect(() => {
+    let interval: NodeJS.Timeout | null = null
+    console.log('EventUpdaterApiOnly::useEffect')
+
+    if (!chainId || pending.length <= 0) {
+      console.log('EventUpdaterApiOnly::exiting', chainId, pending.length)
+
+      return
+    }
+
+    async function update(): Promise<void> {
+      console.log('EventUpdaterApiOnly::calling update')
+
+      const lastBlockNumber = await library?.getBlockNumber()
+      if (!chainId || !lastBlockNumber) {
+        console.log('EventUpdaterApiOnly::update, exiting', chainId, lastBlockNumber)
+
+        return
+      }
+      const orders = await Promise.all(
```

Comment:
I feel like this is also a bit complex to read, specially if we are in nested function, of the function of a useEffect wich is at the same inside a the updater function 🤯

i wonder if it would make sense to define a helper function that would allow to turn this, into:
```ts
const orders = await Promise.all(
        pending.map<Promise<[Order, OrderMetaData | null]>>(async storeOrder => {
          const apiOrder = await getOrder(chainId, storeOrder.id)
          return [storeOrder, apiOrder]
        })
      )
      console.log('EventUpdaterApiOnly::got api orders', orders)

      const ordersBatchData: OrderLogPopupMixData[] = orders
        .filter(([, apiOrder]) => apiOrderIsSettled(apiOrder))
        .map(([orderFromStore, orderFromApi]) => {
          const summary = _computeFulfilledSummary({ orderFromStore, orderFromApi })
          return {
            id: orderFromStore.id,
            fulfillmentTime: new Date().toISOString(),
            // TODO: get this from the API!!!
            transactionHash: 'nothing for now',
            summary
          }
        })
```

Assuming we have an function `_ordersBatchData` we could reduce the code into:
```ts
let orders = await Promise.all(
    pending.map(_ordersBatchData)
)

// orders would be null or the data, so we can filter just the nulls
orders = orders.filter(Boolean)
```

Reply:
does anyone know on a performance level if `.map > .filter` or any combination of looping array prototype functions is faster than a single `.reduce` ? just curious.